### PR TITLE
Collect performance metrics from e2e tests in prow cluster

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:

When running e2e tests in prow, the seed prometheus pushes selected metrics from gardenlet to a dedicated prometheus instance running in the prow build cluster (ref https://github.com/gardener/ci-infra/pull/2807).

For this, provider-local injects a remote write config into the seed prometheus to `http://prometheus-performance.prow.gardener.cloud.local:9090/api/v1/write` when running in CI. This hostname is rewritten to the in-cluster service name in the prow cluster's coredns.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener-community/hackathon/tree/main/2024-12_Schelklingen

**Special notes for your reviewer**:

This PR is just about introducing a mechanism for persisting performance metrics of the e2e tests over time for further analysis. There are no such performance metrics yet, but they can be added later on in a dedicated PR.
For now, we only push the `up` metric as a proof of concept.

Later on, we can observe, for example, how the cluster creation time evolves with changes in this repository. For this, we only need to add the metrics to gardenlet's metrics registry and allow-listing them in the remote write endpoint.

In draft until
- [x] we have verified this on the prow cluster (verified together with @oliver-goetz)

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/provider-local/webhook/prometheus/options.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	ExternalLabels  MapFlag
+
+	config *WebhookConfig
+}
+
+// AddFlags implements Flagger.AddFlags.
+func (w *WebhookOptions) AddFlags(fs *pflag.FlagSet) {
+	// if w.ExternalLabels == nil {
+	// 	w.ExternalLabels = MapFlag{}
+	// }
```
Comment (LucaBernstein) [inline pkg/provider-local/webhook/prometheus/options.go]: Is this to be activated at a later point in time? Is the functionality working with this commented-out block but nil ExternalLabels?

**Review by LucaBernstein — COMMENTED** [review]
Only one minor question, rest:
/lgtm

File: pkg/provider-local/webhook/prometheus/options.go (hunk 1 above)
Comment (timebertt) [reply pkg/provider-local/webhook/prometheus/options.go]: Good point, I forgot to clean this up. Removed the pointer from the `MapFlag` methods and un-commented the proper initialization.

**Review by LucaBernstein — COMMENTED** [review]
Thank you!
/lgtm

**Comment by timebertt** [comment]
Thanks for the review @LucaBernstein. Let's get this in.
/approve

**Pull request merged**